tests/unit/lfp/lfp.cpp

```diff
@@ -0,0 +1,104 @@
+#define BOOST_TEST_MODULE LFPTest
+#define BOOST_TEST_MAIN
+
+#include <iostream>
+
+#include <boost/test/unit_test.hpp>
+#include <boost/test/test_case_template.hpp>
+
+#include "coreneuron/io/lfp.hpp"
+#include "coreneuron/mpi/nrnmpi.h"
+
+template <typename F>
+double integral(F f, double a, double b, int n) {
+    double step = (b - a) / n;  // width of each small rectangle
+    double area = 0.0;          // signed area
+    for (int i = 0; i < n; i++) {
+        area += f(a + (i + 0.5) * step) * step;  // sum up each small rectangle
+    }
+    return area;
+}
+
+std::array<double, 3> operator*(const double x, const std::array<double, 3>& y) {
+    return {x * y[0], x * y[1], x * y[2]};
+}
+
+std::array<double, 3> operator+(const std::array<double, 3>& x, const std::array<double, 3>& y) {
+    return {x[0] + y[0], x[1] + y[1], x[2] + y[2]};
+}
+
+BOOST_AUTO_TEST_CASE(LFP_PointSource_LineSource) {
+    using namespace coreneuron;
+#if NRNMPI
+    nrnmpi_init(nullptr, nullptr);
+#endif
+    double segment_length{1.0e-6};
+    double segment_start_val{1.0e-6};
+    std::array<double, 3> segment_start = std::array<double, 3>{0.0, 0.0, segment_start_val};
+    std::array<double, 3> segment_end = segment_start +
+                                        std::array<double, 3>{0.0, 0.0, segment_length};
+    double floor{1.0e-6};
+    double pi{3.141592653589};
```

Comment:
`PI` is passed as global variable by neuron. Probably you can use it

Reply:
Ok. Thanks.